### PR TITLE
feat: add exportable internal functions

### LINKED ISSUE #491: Exporting functions from submodules
Right now, there is no simple way to export a function that was imported from a submodule. This makes it hard to modularize an assembly contract, therefore complicating the whole developing process. 

I would love if we could have the option to make:

> import {f1, f2, f3 ... fn} from './bar'} 
> export {f1, f2, f3 ... fn}

### PR DESCRIPTION
**Pull request by willemneal**
fixes #491

Submodule functions decorated with `@nearBindgen` are wrapped in the same manner as those exported from the entry file.  This also helps to signal that this function will be a module export to the reader.

For example,

`index.ts`
```ts
export { foo} from "./foo"
```

`foo.ts`
```ts
// @ts-ignore
@nearBindgen
export function foo(): string {
  return "hello";
}
```

Unfortunately Typescript doesn't allow decorators on

### CODE REVIEW

File: simulator/assembly/__tests__/empty.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+// @nearfile out
```
Comment (willemneal) [inline simulator/assembly/__tests__/empty.ts:1]:
```suggestion
```

File: simulator/__tests__/cross.spec.ts [hunk 2]
```diff
@@ -65,6 +65,7 @@ describe("cross contract calls", () => {
 
   test("getting view with arg", () => {
     const res = words.view("reverse", { word: { text: "hello" } });
+    console.log(JSON.stringify(res, null, 2));
```
Comment (mehtaphysical) [inline simulator/__tests__/cross.spec.ts:68]: nit: extra `console.log`

File: bindgen/src/JSONBuilder.ts [hunk 3]
```diff
@@ -101,19 +106,33 @@ export class JSONBindingsBuilder extends BaseVisitor {
...
+  needsWrapper(node: FunctionDeclaration): boolean {
+    let isExport = node.is(CommonFlags.EXPORT);
     let alreadyWrapped = this.wrappedFuncs.has(toString(node.name));
-    let isNotExport = !node.is(CommonFlags.EXPORT);
     let noInputOrOutput = numOfParameters(node) == 0 && returnsVoid(node);
-    if (isNotEntry || alreadyWrapped || isNotExport || noInputOrOutput) {
+    if (
+      !isExport ||
+      alreadyWrapped ||
+      noInputOrOutput ||
```
Comment (mehtaphysical) [inline bindgen/src/JSONBuilder.ts:127]: nice! just realizing now that:

```ts
export { myFunc } from './path.ts'
```

was always possible as long as `myFunc` han to inputs or outputs.

**Review by mehtaphysical — APPROVED** [review]
Looks good! Cool feature.

File: simulator/__tests__/cross.spec.ts (hunk 2 above)
Comment (willemneal) [reply simulator/__tests__/cross.spec.ts:68]: Thanks! Next time can you use the suggestion feature?

Comment (willemneal) [reply simulator/__tests__/cross.spec.ts:68]:
```suggestion
```

File: simulator/__tests__/cross.spec.ts [hunk 4]
```diff
@@ -90,4 +91,12 @@ describe("cross contract calls", () => {
       )
     ).toBe(true);
   });
+
+  test("exported from submodule", () => {
+    const str = "hello";
+    const res = words.view("capitalize", { s: str}).result.outcome as any;
+    console.log(JSON.stringify(res, null, 2));
```
[the remark this answers is not in the document]
Comment (willemneal) [reply simulator/__tests__/cross.spec.ts:98]:
```suggestion
```

**Pull request merged**